File: ImperatorToCK3/Source/Mappers/TraitMapper/TraitMapping.cpp
#include "TraitMapping.h"
#include "ParserHelpers.h"
#include "CommonRegexes.h"

mappers::TraitMapping::TraitMapping(std::istream& theStream)
{
	registerKeyword("ck3", [this](std::istream& theStream) {
		ck3Trait = commonItems::getString(theStream);
	});
	registerKeyword("imp", [this](std::istream& theStream) {
		impTraits.insert(commonItems::getString(theStream));
	});
	registerMatcher(commonItems::catchallRegexMatch, commonItems::ignoreItem);

	parseStream(theStream);
	clearRegisteredKeywords();
}

File: ImperatorToCK3/Mappers/Trait/TraitMapping.cs
using commonItems;
using System.Collections.Generic;

namespace ImperatorToCK3.Mappers.Trait;

public sealed class TraitMapping {
	public SortedSet<string> ImperatorTraits { get; } = new();
	public string? CK3Trait { get; set; }

	public TraitMapping(BufferedReader mappingReader) {
		var parser = new Parser();
		parser.RegisterKeyword("ck3", reader => CK3Trait = reader.GetString());
		parser.RegisterKeyword("ir", reader => ImperatorTraits.Add(reader.GetString()));
		parser.RegisterRegex(CommonRegexes.Catchall, ParserHelpers.IgnoreAndLogItem);
		parser.ParseStream(mappingReader);
		parser.ClearRegisteredRules();
	}
}